TheMET/TheMET/Network/ImageLoader.swift

```diff
@@ -10,24 +10,33 @@ import UIKit
 
 class ImageLoader {
     
+    private var loadedImagesCollection: [URL : UIImage] = [:]
+    
     func loadImage(urlString: String, completion: @escaping (UIImage?) -> Void) {
+        
         guard let imageURL = URL(string: urlString) else {
             completion(nil)
             return
         }
-        DispatchQueue.global().async {
-            let executeComplitionOnMain: (UIImage?) -> Void = { image in
-                DispatchQueue.main.async {
-                    completion(image)
+        if self.loadedImagesCollection.keys.contains(imageURL) {
```

Comment:
1) Instead of checking for the key, check for the value inside the dictionary using `if-let`
2) try to use `guard` for this check to not affect the rest of the code